### PR TITLE
Add a planet menu entry to create a colony building

### PR DESCRIPTION
**Pull request by vincele**
Avoid uninhabitable planets

Signed-off-by: Vincent Legoll vincent.legoll@gmail.com

### CODE REVIEW

**Comment by vincele** [comment]
Was previously https://github.com/freeorion/freeorion/pull/167

**Comment by Dilvish-fo** [comment]
Looks like you've been making so many mods to the ObjectWnd you're having trouble keeping track of what you need to put into the branch for the PR...

```
freeorion/UI/ObjectListWnd.cpp:2470:104: error: ‘GetColonizerSpecies’ was not declared in this scope

```

File: UI/ObjectListWnd.cpp [hunk 1]
```diff
@@ -2356,12 +2356,37 @@ void ObjectListWnd::ObjectRightClicked(GG::ListBox::iterator it, const GG::Pt& p
...
+
     std::map<std::string, int> all_foci;
     UniverseObjectType type = obj->ObjectType();
     if (type == OBJ_PLANET) {
         menu_contents.next_level.push_back(GG::MenuItem(UserString("SP_PLANET_SUITABILITY"), 2, false, false));
 
+        TemporaryPtr<Planet> planet = GetPlanet(object_id);
+        // My planet is empty, we may now want to create a colony there...
+        if (planet->OwnedBy(app->EmpireID()) && planet->GetMeter(METER_POPULATION)->Current() == 0.0f) {
+                if (Empire* emp = GetEmpire(app->EmpireID())) {
```
Comment (geoffthemedio) [inline UI/ObjectListWnd.cpp:2371]: why 8-space indent here?

File: UI/ObjectListWnd.cpp [hunk 2]
```diff
@@ -2432,11 +2461,27 @@ void ObjectListWnd::ObjectRightClicked(GG::ListBox::iterator it, const GG::Pt& p
...
+                std::set<std::string>::const_iterator it = species.begin();
+                std::advance(it, id - MENUITEM_CREATE_COLONY_BASE - 1);
+
+                std::string species_name = *it; // SP_ABADDONI
+                std::string building_name = "BLD_COL_" + species_name.substr(3); // BLD_COL_ABADDONI
+
+                if (!empire->ProducibleItem(BT_BUILDING, building_name, object_id)) {
+                    ErrorLogger() << "Cannot produce building at that location";
+                } else {
+                app->Orders().IssueOrder(OrderPtr(
```
Comment (geoffthemedio) [inline UI/ObjectListWnd.cpp:2481]: indent within braces

**Comment by geoffthemedio** [comment]
In general, this code could use more comments explaining what it's doing

**Comment by vincele** [comment]
@Dilvish-fo : yes I'm having a full stack of patches going for a long time now, it was quilt on svn, then I switched to stg inside a git branch. it's quite nice, but obviously you have to keep the big picture in mind when you submit parts of it. I did not compile-test this PR. My bad. I'll do it next time.

@all : Thanks for the feedback, I'll respin the patches after taking care of it.

**Pull request closed without merging**